### PR TITLE
stat/distuv: Improve Exponential test coverage and factor out some tests.

### PR DESCRIPTION
**Pull request by romanwerpachowski**
- Added tests to `exponential_test.go`
- Factored out some tests to `general_test.go` so that they're used for other distributions which are `ConjugateUpdater`
- fixed `Norm` so that it passes the new tests

Please take a look.

### CODE REVIEW

File: stat/distuv/exponential_test.go [hunk 1]
```diff
@@ -79,6 +79,48 @@ func testExponential(t *testing.T, dist Exponential, i int) {
...
+	}
+
+	deriv := make([]float64, 1)
+	dist.Score(deriv, -0.0001)
+	if deriv[0] != 0 {
+		t.Errorf("Score is not {0} for a negative argument. Got %v", deriv[0])
+	}
+	dist.Score(deriv, 0)
+	if !math.IsNaN(deriv[0]) {
+		t.Errorf("Score is not {NaN} at 0. Got %v", deriv[0])
```
Comment (kortschak) [inline stat/distuv/exponential_test.go:115]: The braces aren't necessary here.

File: stat/distuv/general_test.go [hunk 2]
```diff
@@ -182,7 +213,10 @@ func testDerivParam(t *testing.T, d derivParamTester) {
 	for _, v := range quantiles {
 		d.setParameters(initParams)
 		x := d.Quantile(v)
-		d.Score(deriv, x)
+		returnedDeriv := d.Score(deriv, x)
```
Comment (kortschak) [inline stat/distuv/general_test.go:216]: s/returnedDeriv/gotDeriv/g

File: stat/distuv/norm.go [hunk 3]
```diff
@@ -48,6 +48,12 @@ func (n Normal) CDF(x float64) float64 {
 func (n *Normal) ConjugateUpdate(suffStat []float64, nSamples float64, priorStrength []float64) {
 	// TODO: Support prior strength with math.Inf(1) to allow updating with
 	// a known mean/standard deviation
+	if len(suffStat) != 2 {
```
Comment (kortschak) [inline stat/distuv/norm.go:51]: `if len(suffStat) != n.NumSuffStat() {` and update the documentation to that effect.

File: stat/distuv/norm.go [hunk 4]
```diff
@@ -48,6 +48,12 @@ func (n Normal) CDF(x float64) float64 {
...
+		panic("norm: incorrect suffStat length")
+	}
+	if len(priorStrength) != 2 {
```
Comment (kortschak) [inline stat/distuv/norm.go:54]: `if len(priorStrength) != n.NumSuffStat() {` and update the documentation to that effect.

**Comment by romanwerpachowski** [comment]
Thanks for the review. PTAL.

**Pull request merged**